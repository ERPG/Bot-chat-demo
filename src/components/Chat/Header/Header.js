import React from 'react';
import './Header.css';

/** The header of the Chat */

const Header = () => {
  return (
  <div>
    <div className="landbot-header" data-landbot="header-container">
      <h1 className="subtitle" data-landbot="header-subtitle">Landbot</h1>
    </div>
  </div>
  )
}

export default Header;