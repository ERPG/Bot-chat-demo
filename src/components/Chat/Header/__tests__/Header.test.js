import React from 'react';
import { render } from '@testing-library/react';
import Header from '../Header';
import { customRender } from '../../../../../__tests__/__utils__/testUtils';

describe('Header component', () => {
  const headerContainerSelector = 'header-container';
  const headerTextSelector = 'header-subtitle';

  it('Should render component without errors', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  it('Should have a container', () => {
    const { getByDataLb } = customRender(<Header />);
    const element = getByDataLb(headerContainerSelector)
    expect(element).toBeDefined();
  });

  it('Should have a text', () => {
    const { getByDataLb } = customRender(<Header />);
    const element = getByDataLb(headerTextSelector);
    expect(element.textContent).toBe('Landbot');
  });
});