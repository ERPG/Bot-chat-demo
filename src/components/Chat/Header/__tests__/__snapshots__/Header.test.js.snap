// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should render component without errors 1`] = `
<div>
  <div>
    <div
      class="landbot-header"
      data-landbot="header-container"
    >
      <h1
        class="subtitle"
        data-landbot="header-subtitle"
      >
        Landbot
      </h1>
    </div>
  </div>
</div>
`;
